app/models/concerns/downloadable.rb

```diff
@@ -1,28 +1,28 @@
 module Downloadable
   extend ActiveSupport::Concern
 
-  def source_csv_file_count
-    Rational(total_count, source_csv_limit).ceil
+  def csv_limit
+    self.class::CSV_LIMIT
   end
 
-  def source_csv_limit
-    self.class::SOURCE_CSV_LIMIT
+  def csv_pagination(file: 1)
+    { page: file, per_page: csv_limit }
   end
 
-  def source_csv_pagination(file: 1)
-    { page: file, per_page: source_csv_limit }
+  def csv_file_count
+    Rational(total_count, csv_limit).ceil
   end
 
-  def source_dataset(file: 1)
+  def csv_dataset(file: 1)
```

Comment:
I thought you wanted to leave this one as source_dataset as it is the data for the table as well as the csv?

Reply:
I did initially, but on reflection it is internally bound to csv via pagination so I thought better to simplify the names.